component/kubelet.jsonnet

```
@@ -11,9 +11,7 @@ local kubeletConfigs = [
     metadata+: {
       labels+: common.DefaultLabels,
     },
-    spec: params.kubeletConfigs[nodeGroup] {
-      assert !std.objectHas(self.kubeletConfig, 'maxPods') || self.kubeletConfig.maxPods <= 110 : 'kubeletConfig.maxPods cannot be greater than 110',
-    },
+    spec: params.kubeletConfigs[nodeGroup],
```

We may want to print a warning if users go above the best-practice upstream K8s value of 110 (cf. https://kubernetes.io/docs/setup/best-practices/cluster-large/). You could use something like

```jsonnet
local checkMaxPods(config) =
  if
    std.objectHas(config.kubeletConfig, 'maxPods')
    && config.kubeletConfig.maxPods > 110
  then
    config {
      kubeletConfig: {
        maxPods: std.trace(
          '[WARNING] Upstream Kubernetes recommends to have maximum pods per node <= 110.',
          config.kubeletConfig.maxPods
        ),
      },
    }
  else
    config;
```

and 

```suggestion
    spec: checkMaxPods(params.kubeletConfigs[nodeGroup]),
```

---

I was musing about this as well. My first hunch was: nobody will care about warnings. But it might be a good thing anyway.

@simu Do we have the tools to test such warnings?